examples/custom-widget-with-1st-party-data/src/controller/controller.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import { client } from '../client';
+import { setupCustomWidget } from './custom-widget';
+
+/**
+ * This component renders the main controller.
+ * The main controller responds to the initial handshake from Datadog and sets up any App-wide behavior.
+ *
+ * @see https://github.com/DataDog/apps/blob/-/docs/en/programming-model.md#main-controller-iframe
+ */
+function Controller(): JSX.Element {
+    React.useEffect(() => {
+        setupCustomWidget(client);
+    }, []);
+
+    return (
+        <>
+            <div>The application controller is running in the background.</div>
+            <a href="http://localhost:3000/custom-widget">
+                Click here to open your custom widget
+            </a>
+        </>
+    );
+}
```

Comment:
I love this. Even though the controller doesn't need to be react, making it react anyways for consistency is great